pinot-tools/src/test/java/org/apache/pinot/tools/realtime/provisioning/MemoryEstimatorTest.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.tools.realtime.provisioning;
+
+import java.io.File;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.io.FileUtils;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.utils.JsonUtils;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+
+public class MemoryEstimatorTest {
+
+  @Test
+  public void testSegmentGenerator() throws Exception {
+
+    // arrange inputs
+    File schemaFile = readFile("memory_estimation/schema.json");
+    File dataCharacteristicsFile = readFile("memory_estimation/data-characteristics.json");
+    File tableConfigFile = readFile("memory_estimation/table-config.json");
+    TableConfig tableConfig = JsonUtils.fileToObject(tableConfigFile, TableConfig.class);
+
+    // act
+    MemoryEstimator.SegmentGenerator
+        segmentGenerator = new MemoryEstimator.SegmentGenerator(dataCharacteristicsFile, schemaFile, tableConfig, true);
+    File generatedSegment = segmentGenerator.generate();
+
+    // assert
+    Path metadataFile = Paths.get(generatedSegment.getPath(), "v3", "metadata.properties");
```

Comment:
how can we check for average length for string types?

Reply:
I'm not sure! Do you have any suggestion? Although StringGenerator is a simple class and we're ok not checking that.